src/main/scala/cromwell/engine/db/slick/ExecutionComponent.scala

```diff
@@ -25,21 +25,43 @@ trait ExecutionComponent {
     override def * = (workflowExecutionId, callFqn, status, executionId.?) <>
       (Execution.tupled, Execution.unapply)
 
-    def workflow = foreignKey(
+    def workflowExecution = foreignKey(
       "FK_EXECUTION_WORKFLOW_EXECUTION_ID", workflowExecutionId, workflowExecutions)(_.workflowExecutionId)
 
     def uniqueKey = index("UK_EX_WORKFLOW_EXECUTION_ID",
       (workflowExecutionId, callFqn), unique = true)
   }
 
-  val executions = TableQuery[Executions]
+  protected val executions = TableQuery[Executions]
 
   val executionsAutoInc = executions returning executions.
     map(_.executionId) into ((a, id) => a.copy(executionId = Some(id)))
 
-  val executionByID = Compiled(
-    (id: Rep[Int]) => for {
+  val executionCallFqnsAndStatusesByWorkflowExecutionId = Compiled(
+    (workflowExecutionId: Rep[Int]) => for {
       execution <- executions
-      if execution.executionId === id
+      if execution.workflowExecutionId === workflowExecutionId
+    } yield (execution.callFqn, execution.status))
+
+  val executionsByWorkflowExecutionUuidAndCallFqn = Compiled(
+    (workflowExecutionUuid: Rep[String], callFqn: Rep[String]) => for {
+      execution <- executions
+      if execution.callFqn === callFqn
+      workflowExecution <- workflowExecutions
+      if workflowExecution.workflowExecutionUuid === workflowExecutionUuid
     } yield execution)
+
+  val executionStatusesByExecutionId = Compiled(
+    (executionId: Rep[Int]) => for {
+      execution <- executions
+      if execution.executionId === executionId
+    } yield execution.status)
+
+  // see workflowExecutionsByStatuses
+  def executionStatusesByWorkflowExecutionIdAndCallFqns(workflowExecutionId: Int, callFqns: Traversable[String]) = {
```

Comment:
These projections read pretty strangely, grabbing all the statuses corresponding to the query parameters but not obviously tying them back to the underlying rows, but it looks like that's how updates are done in Slick 3...